Detectors/CTP/Raw/test/CTPtestRawWriter.cxx

```diff
@@ -0,0 +1,173 @@
+
+#define BOOST_TEST_MODULE Test RawReaderWriter class
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_DYN_LINK
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <TRandom.h>
+#include <boost/test/unit_test.hpp>
+#include "Steer/InteractionSampler.h"
+#include "DetectorsRaw/HBFUtils.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "DetectorsRaw/RawFileWriter.h"
+#include "DetectorsRaw/SimpleRawReader.h"
+#include "DetectorsRaw/SimpleSTF.h"
+#include "CommonConstants/Triggers.h"
+#include "Framework/Logger.h"
+#include "Framework/InputRecord.h"
+#include "DPLUtils/DPLRawParser.h"
+#include "CommonUtils/StringUtils.h"
+
+// @brief test and demo for RawFileReader and Writer classes
+// @author ruben.shahoyan@cern.ch
+
+namespace o2
+{
+using namespace o2::raw;
+using namespace o2::framework;
+using IR = o2::InteractionRecord;
+using RDHAny = header::RDHAny;
+constexpr int NCRU = 1;        // numbaer of CRUs, the last one is a special CRU with preformatted data filled
+constexpr int NLinkPerCRU = 2; // number of links per CRU
+// sizes for preformatted pages filling (RDH size will be subtracted from the payload) in the last special CRU
+constexpr std::array<int, NLinkPerCRU> SpecSize = {512, 1024, 8192, 8192};
+constexpr int NPreformHBFPerTF = 32; // number of HBFs with preformatted input per HBF for special last CRU
+const std::string PLHeader = "HEADER          ";
+const std::string PLTrailer = "TRAILER         ";
+const std::string HBFEmpty = "EMPTY_HBF       ";
+const std::string CFGName = "testRawReadWrite";
+
+int nPreformatPages = 0;
+//
+// ========================= simple detector data writer ================================
+//
+struct CTPTestRawWriter { // simple class to create detector payload for multiple links
+
+  RawFileWriter writer{"CTP"};
+  std::string configName = "rawConf.cfg";
+
+  //_________________________________________________________________
+  CTPTestRawWriter(o2::header::DataOrigin origin = "TST", bool isCRU = true, const std::string& cfg = "rawConf.cfg") : writer(origin, isCRU), configName(cfg) {}
+
+  //_________________________________________________________________
+  void init()
+  {
+    // init writer
+    writer.useRDHVersion(6);
+    int feeIDShift = writer.isCRUDetector() ? 8 : 9;
```

Comment:
You should tell if the writer serves in "CRU" mode or "C-RORC". I guess it is CRU-like, as it continuously provides a data.

Reply:
yes, we are cru. But actually I did not want to commit this file, it was very first attempt. I will remove file.